src/esp/sensor/VisualSensor.cpp

```diff
@@ -3,15 +3,60 @@
 // LICENSE file in the root directory of this source tree.
 
 #include "VisualSensor.h"
+#include <Magnum/EigenIntegration/Integration.h>
 
 #include <utility>
 
 #include "esp/gfx/RenderTarget.h"
 
 namespace esp {
 namespace sensor {
-VisualSensor::VisualSensor(scene::SceneNode& node, SensorSpec::ptr spec)
-    : Sensor{node, std::move(spec)}, tgt_{nullptr} {}
+
+VisualSensorSpec::VisualSensorSpec()
+    : SensorSpec(),
+      ortho_scale(0.1f),
+      resolution({84, 84}),
+      encoding("rgba_uint8"),
+      gpu2gpuTransfer(false) {
+  sensorType = SensorType::Color;
+}
+
+void VisualSensorSpec::sanityCheck() {
+  bool isVisualSensor =
```

Comment:
Maybe you can have a function `isVisualSensorSpec`, expose it via pybinding, and apply lit when you create sensor configs in the python code.